src/main.rs

```diff
@@ -429,58 +431,73 @@ Click on a crate to the right to display its build log
 <div class="crates">
 "#;
 
-fn write_output(crates: &[Crate]) {
-    let mut output = File::create(".crates.json").unwrap();
+fn write_output(crates: &[Crate]) -> Result<()> {
+    let mut output = File::create(".crates.json")?;
 
     for c in crates {
-        writeln!(output, "{}", serde_json::to_string(c).unwrap()).unwrap();
+        writeln!(output, "{}", serde_json::to_string(c)?)?;
     }
 
-    fs::rename(".crates.json", "crates.json").unwrap();
+    fs::rename(".crates.json", "crates.json")?;
 
-    let mut output = File::create(".index.html").unwrap();
-    writeln!(output, "{}", OUTPUT_HEADER).unwrap();
+    let mut output = File::create(".index.html")?;
+    writeln!(output, "{}", OUTPUT_HEADER)?;
     for c in crates {
         write!(
             output,
             "<div class=\"row\" onclick=\"change_log(&quot;{}&quot;, &quot;{}&quot;)\"><div class=\"crate\">{} {}</div>",
             c.name, c.version, c.name, c.version
         )
-        .unwrap();
-        write!(output, "<div class=\"status\">").unwrap();
+        ?;
+        write!(output, "<div class=\"status\">")?;
         match &c.status {
             Status::Unknown => write!(output, "Unknown"),
             Status::Passing => write!(output, "Passing"),
             Status::Error(cause) => write!(output, "Error: {}", cause),
-            Status::UB { cause, .. } => write!(output, "UB: {}", cause),
-        }
-        .unwrap();
-        writeln!(output, "</div></div>").unwrap();
+            Status::UB { cause: causes, .. } => {
+                write!(output, "UB: ")?;
+                for cause in causes {
+                    write!(output, "{}", cause.kind)?;
+                    if let Some(source_crate) = &cause.source_crate {
+                        write!(output, "{source_crate}")?;
+                    }
+                }
```

Comment:
This mashes all the causes and source crates together, like this `UB: uninitialized memoryhttp-0.2.6`. We should have a delimiter or something in here, to get something similar to the `join` that I started with.